src/libxtp/bulkesp.cc

```diff
@@ -0,0 +1,768 @@
+//                    }
+                    if (distSq <= acceptDist * acceptDist * scale * scale) {
+                        Bond nb; //bond goes
+                        nb.a = (*i); //from a
+                        nb.b = (*j); //to b
+                        nb.ba = v; //and has a (PBC corrected) vector ba (from b to a))
+                        nb.a_indx = i - _atoms.begin();
+                        nb.b_indx = j - _atoms.begin();
+                        bonds.push_back(nb);
+                    }
+                }
+            }
+            CTP_LOG(ctp::logDEBUG, *_log) << " BreakIntoMolecules(): " << bonds.size() << " bonds found.\n" << flush;
+            //        for (std::list<Bond>::iterator b = bonds.begin(); b != bonds.end(); ++b){
+            //            CTP_LOG(ctp::logDEBUG, *_log) << b->a_indx <<" - "<<b->b_indx<<"\t"<<b->ba<<"\t"<<sqrt(b->ba*b->ba)<< flush;
+            //        }
+
+
+            //now add all the atoms that have no bonds as a separate molecule
+            Bulkesp::Molecule leftover;
+            for (vector<ctp::QMAtom*>::iterator i = _atoms.begin(); i != _atoms.end(); ++i) {
+                bool found = false; //does this atom have bonds?
+                for (std::list<Bond>::iterator b = bonds.begin(); b != bonds.end(); ++b) {
+                    if (b->a == *i || b->b == *i) {
+                        found = true;
+                        break;
+                    }
+                }
+                if (!found) //atom has no bonds
+                    leftover.atoms.push_back(*i);
+                leftover.atomIndeces.push_back(i - _atoms.begin());
+            }
+            if (leftover.atoms.size() > 0) {
+                mols.push_back(leftover);
+                CTP_LOG(ctp::logDEBUG, *_log) << " BreakIntoMolecules(): put " << leftover.atoms.size()
+                        << " Unbonded atoms into molecule " << mols.size() - 1 << ";\t" << bonds.size() << " bonds left.\nThey were:" << flush;
+                for(std::vector<ctp::QMAtom*>::iterator it = leftover.atoms.begin(); it != leftover.atoms.end(); ++it) {
+                    /* std::cout << *it; ... */
+                    CTP_LOG(ctp::logDEBUG, *_log) << (*it)->type << "\t" << (*it)->x << "\t" << (*it)->y << "\t" << (*it)->z << endl << flush;
+                }
+                exit(-1);
+                
+            }
+
+
+
+            //assign bonds to molecules
+            while (bonds.size() > 0) {
+                //init the new molecule with the last bond we have
+                Bulkesp::Molecule m;
+                std::list<Bond>::iterator last = bonds.end();
+                std::advance(last, -1);
+                m.atoms.push_back(last->a);
+                m.atomIndeces.push_back(last->a_indx);
+                leftover.atomIndeces.push_back(bonds.end()->a_indx);
+                if (periodic) {//unwrap system by moving b
+                    last->b->x = last->a->x - last->ba.getX();
```

Comment:
why not use the vector formulation?

Reply:
Written before vector formulation was fully implemented. Will fix.